msal/tests/Test.MSAL.NET.Unit.net45/UnifiedCacheFormatTests.cs

```diff
@@ -0,0 +1,308 @@
+            Assert.AreEqual(ExpectedAtCacheKeyIosGeneric, key.GetiOSGenericKey());
+        }
+
+        private void ValidateRt(TokenCache cache)
+        {
+            ValidateCacheEntityValue
+                (ExpectedRtCacheValue, cache.GetAllRefreshTokenCacheItems(requestContext));
+
+            var rtCacheItem = cache.GetAllRefreshTokensForClient(requestContext).First();
+            var key = rtCacheItem.GetKey();
+
+            Assert.AreEqual(ExpectedRtCacheKey, key.ToString());
+
+            Assert.AreEqual(ExpectedRtCacheKeyIosService, key.GetiOSServiceKey());
+            Assert.AreEqual(ExpectedRtCacheKeyIosAccount, key.GetiOSAccountKey());
+            Assert.AreEqual(ExpectedRtCacheKeyIosGeneric, key.GetiOSGenericKey());
+        }
+
+        private void ValidateIdToken(TokenCache cache)
+        {
+            ValidateCacheEntityValue
+                (ExpectedIdTokenCacheValue, cache.GetAllIdTokenCacheItems(requestContext));
+
+            var idTokenCacheItem = cache.GetAllIdTokensForClient(requestContext).First();
+            var key = idTokenCacheItem.GetKey();
+
+            Assert.AreEqual(ExpectedIdTokenCacheKey, key.ToString());
+
+            Assert.AreEqual(ExpectedIdTokenCacheKeyIosService, key.GetiOSServiceKey());
+            Assert.AreEqual(ExpectedIdTokenCacheKeyIosAccount, key.GetiOSAccountKey());
+            Assert.AreEqual(ExpectedIdTokenCacheKeyIosGeneric, key.GetiOSGenericKey());
+        }
+
+        private void ValidateAccount(TokenCache cache)
+        {
+            ValidateCacheEntityValue
+                (ExpectedAccountCacheValue, cache.GetAllAccountCacheItems(requestContext));
+
+            var accountCacheItem = cache.GetAllAccounts(requestContext).First();
+            var key = accountCacheItem.GetKey();
+
+            Assert.AreEqual(ExpectedAccountCacheKey, key.ToString());
+
+            Assert.AreEqual(ExpectedAccountCacheKeyIosService, key.GetiOSServiceKey());
+            Assert.AreEqual(ExpectedAccountCacheKeyIosAccount, key.GetiOSAccountKey());
+            Assert.AreEqual(ExpectedAccountCacheKeyIosGeneric, key.GetiOSGenericKey());
+        }
+
+        private void ValidateCacheEntityValue(string expectedEntityValue, ICollection<string> entities)
+        {
+            Assert.IsTrue(entities.Count == 1);
+
+            var actualPayload = JsonConvert.DeserializeObject<JObject>(entities.First());
+            var expectedPayload = JsonConvert.DeserializeObject<JObject>(expectedEntityValue);
+
+            Assert.AreEqual(expectedPayload.Count, actualPayload.Count);
+
+            foreach (KeyValuePair<string, JToken> prop in expectedPayload)
+            {
+                var propName = prop.Key;
+                var expectedPropValue = prop.Value;
+                var actualPropValue = actualPayload.GetValue(propName);
+
+                Assert.AreEqual(expectedPropValue, actualPropValue);
+            }
+        }
+    }
+}
+#endif
```

Comment:
this is pretty cool.  What about ADFS scenarios?

Reply:
can be also included quite easily, main challenge is to create test data for adfs -> sts response and expectations , it probably should be done in separate PR related to enabling ADFS in msal